src/ui/wxWidgets/PasswordSafeFrame.cpp

```diff
@@ -684,6 +706,25 @@ void PasswordSafeFrame::CreateMenubar()
   // Update menu selections
   GetMenuBar()->Check((IsTreeView()) ? ID_TREE_VIEW : ID_LIST_VIEW, true);
   GetMenuBar()->Check(PWSprefs::GetInstance()->GetPref(PWSprefs::UseNewToolbar) ? ID_TOOLBAR_NEW : ID_TOOLBAR_CLASSIC, true);
+  UpdateTreeSortMenu();
+}
+
+/**
+ * Update check for sorting and enable or disable the menu bar for orting
+ */
+void PasswordSafeFrame::UpdateTreeSortMenu()
+{
+  auto menuBar = GetMenuBar();
+  
+  menuBar->Check(ID_SORT_TREE_BY_GROUP, IsTreeSortGroup() ? true : false);
+  menuBar->Check(ID_SORT_TREE_BY_NAME, IsTreeSortName() ? true : false);
+  menuBar->Check(ID_SORT_TREE_BY_DATE, IsTreeSortDate() ? true : false);
+
+  menuBar->Enable(ID_SORT_TREE_MENU, IsTreeView() ? true : false);
+  menuBar->Enable(ID_SORT_TREE_BY_GROUP, IsTreeView() ? true : false);
+  menuBar->Enable(ID_SORT_TREE_BY_NAME, IsTreeView() ? true : false);
+  menuBar->Enable(ID_SORT_TREE_BY_DATE, IsTreeView() ? true : false);
```

Comment:
Is this still needed if already handled by OnUpdateUI?

Reply:
You are right, 4 lines removed.